src/surface_conditions/surface_conditions.jl

```diff
@@ -170,9 +170,14 @@ function surface_state_to_conditions(
     T = if isnothing(sfc_prognostic_temp)
         if isnothing(surf_state.T) && (
             coordinates isa Geometry.LatLongZPoint ||
-            coordinates isa Geometry.LatLongPoint
+            coordinates isa Geometry.LatLongPoint ||
+            atmos.sfc_temperature isa RCEMIPIISST
         )
-            surface_temperature(atmos.sfc_temperature, coordinates)
+            surface_temperature(
+                atmos.sfc_temperature,
+                atmos.model_config,
+                coordinates,
+            )
```

Comment:
`atmos.model_config` carries redundant information (we can find if we are on a Box or on a Sphere from the `Space`) and I would like to eventually get rid of it.

Given this, and the fact that the chain of conditionals in this function started being very complex, could you try rewriting this function so that it is easier to understand what it does and it does not use `model_config`?

Reply:
`eltype` of `coordiantes` will be `LatLongPoint` for a sphere and `XYPoint` for a box.

Reply:
Oh that's a good point. We can dispatch on coordinates then.